api/managementpb/role.proto

```diff
@@ -0,0 +1,116 @@
+syntax = "proto3";
+
+package management;
+
+option go_package = "api/managementpb;managementpb";
+
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "google/api/annotations.proto";
+
+message RoleData {
+  uint32 role_id = 1;
+  string title = 2;
+  string filter = 3;
+}
+
+message RoleID {
+  uint32 role_id = 1;
+}
+
+message EmptyResponse {}
+
+message AssignRoleRequest {
+  uint32 role_id = 1;
+  uint32 user_id = 2;
+}
+
+message ListRolesRequest {}
+
+message ListRolesResponse {
+  repeated RoleData roles = 1;
+}
+
+// Service service provides public methods for managing Services.
+service Role {
+  // CreateRole creates a new role.
+  rpc CreateRole(RoleData) returns (RoleID) {
+    option (google.api.http) = {
+      put: "/v1/management/Role"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Create Role"
+      description: "Creates a new Role."
+    };
+  }
+
+  // UpdateRole creates a new role.
+  rpc UpdateRole(RoleData) returns (EmptyResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/Role"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Update Role"
+      description: "Updates an existing Role."
+    };
+  }
+
+  // DeleteRole creates a new role.
+  rpc DeleteRole(RoleID) returns (EmptyResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/Role/Delete"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Delete Role"
+      description: "Deletes a Role."
+    };
+  }
+
+  // GetRole retrieves a single role.
+  rpc GetRole(RoleID) returns (RoleData) {
```

Comment:
In general I think there are two approaches:

1. Have a request/response message for each service
2. Combine messages and reuse them

I opted for the second approach since I believe it's easier to read, but I'm happy to change it to individual messages if that's the agreed upon convention.

Reply:
I haven't heard about any company-wide conventions, but I find the request/response approach more maintainable and expandable because

- we can still create messages that represent particular objects to reuse them in response/request. 
- the main concern: protobuf doesn't allow to change methods signature without upgrading API version because it's a breaking change, so if there will be a need to add some field to the request/response in the future, the request/response object name may start to contain unrelated fields and thus become confusing. 
 
However I see that the methods which have a high risk of being expanded (ListRoles, AssignRole) are already designed using that request/response model, so if the others doesn't have that risk I guess it's fine to keep the current approach.

Reply:
Great feedback.
I wasn't aware changing the message causes a breaking change. I could not find the information in the docs either, except for a comment in a GitHub issue in grpc repo.
If you have any reference, it would be appreciated.

I'm struggling a bit with not being able to reuse message fields from `RoleData` message in `GetRoleResponse` without nesting it under a field (https://github.com/percona/pmm/blob/bce98381a4c8f57bb5d98a1699a428726de5e295/api/managementpb/role.proto#L57-L61)
Any best practices you recommend?

Reply:
It's configured in CI to fail:
https://github.com/percona/pmm/blob/c1cfb049058a3b01cba3616bb57475e7e24f570c/Makefile.include#L37 

Regarding the reuse problem - if nesting is not an option, then I don't know unfortunately any tricks how to reuse the messages, I thought about smth like
```
 message GetRoleResponse { 
   Role role = 1;
 } 

message Role {
   uint32 role_id = 1; 
   string title = 2; 
   string filter = 3; 
}
```